### DISCUSSION TITLE
Running s3 tests

### DISCUSSION THREAD

**Discussion by fischerscode** [serverpod/serverpod#335] [Q&A] [answered]
I just tried running the `cloud_storage_s3_test` test. And they failed.
After a while, I realized the the access key is invalid ("The AWS Access Key Id you provided does not exist in our records").

Then I started minio as an local s3 storage. But it seems like the `S3CloudStorage.publicHost` is not used by the `AwsS3Uploader`. (Would be nice if `serverpod_cloud_storage_s3` would support 3th party s3 instances.)

Is there a way to test s3 without an aws account?

**Comment by vlidholt** [comment, marked as answer]
Currently these tests are only run by me. They are not used in CI. Will probably try to get a better setup for that in the future!
